### PR TITLE
ceph-pull-requests: use XUnit for publishing the test result of ctest

### PR DESCRIPTION
**Pull request by tchaikov**
for better developer experience

Signed-off-by: Kefu Chai <kchai@redhat.com>

### CODE REVIEW

**Comment by tchaikov** [comment]
it depends on https://github.com/ceph/ceph/pull/40400

**Comment by tchaikov** [comment]
jenkins retest this please.

File: ceph-pull-requests/config/definitions/ceph-pull-requests.yml [hunk 1]
```diff
@@ -91,7 +91,10 @@
                 - shell:
                     !include-raw:
                       - ../../build/kill-tests
-
+      - xunit:
+          types:
+            - ctest:
+                pattern: "build/Testing/**/Test.xml"
```
Comment (epuertat) [inline ceph-pull-requests/config/definitions/ceph-pull-requests.yml:97]: junit has a new interesting feature: it [integrates with Github checks](https://plugins.jenkins.io/junit/), but unfortunately I don't think the XML generated by [ctest is XML JUnit compliant](https://gitlab.kitware.com/cmake/cmake/-/issues/18654).

**Review by epuertat — APPROVED** [review]
Great!

File: ceph-pull-requests/config/definitions/ceph-pull-requests.yml (hunk 1 above)
Comment (tchaikov) [reply ceph-pull-requests/config/definitions/ceph-pull-requests.yml:97]: yeah, i like Github Check APIs --  it integrates better with github, and one less click =)

but, probably we can use some XSL for translating the ctest XML to JUnit XML format. haven't looked into this though.

**Pull request merged**